entry_types/scrolled/package/spec/frontend/MediaPlayer-spec.js

```diff
@@ -30,41 +32,103 @@ describe('MediaPlayer', () => {
   });
 
   it('renders video tag for video type sources', () => {
-    const {container} = render(<MediaPlayer type={'video'} sources={getAudioSources()} />);
+    let state = getInitialPlayerState();
+    const {container} = render(<MediaPlayer type={'video'} sources={getAudioSources()} playerState={state} playerActions={getPlayerActions()} />);
     
     expect(container.querySelector('video')).toBeDefined();
   });
 
   it('do not requests new Player for same sources', () => {
     const spyMedia = jest.spyOn(media, 'getPlayer');
-    const {rerender} = render(<MediaPlayer type={'video'} sources={getVideoSources()} />);
+    let state = getInitialPlayerState();
+    let playerActions = getPlayerActions();
+    const {rerender} = render(<MediaPlayer type={'video'} sources={getVideoSources()} playerState={state} playerActions={playerActions} />);
     expect(spyMedia).toHaveBeenCalledTimes(1);
-    rerender(<MediaPlayer type={'video'} sources={getVideoSources()} />);
+    rerender(<MediaPlayer type={'video'} sources={getVideoSources()} playerState={state} playerActions={playerActions} />);
     expect(spyMedia).toHaveBeenCalledTimes(1);
   });
 
   it('requests new Player on sources change', () => {
     const spyMedia = jest.spyOn(media, 'getPlayer');
-    const {rerender} = render(<MediaPlayer type={'video'} sources={getVideoSources()} />);
+    let state = getInitialPlayerState();
+    let playerActions = getPlayerActions();
+    const {rerender} = render(<MediaPlayer type={'video'} sources={getVideoSources()} playerState={state} playerActions={playerActions} />);
     expect(spyMedia).toHaveBeenCalledTimes(1);
     let videoSources = getVideoSources();
     videoSources[0].src = 'example.mp4';
-    rerender(<MediaPlayer type={'video'} sources={videoSources} />);
+    rerender(<MediaPlayer type={'video'} sources={videoSources} playerState={state} playerActions={playerActions} />);
     expect(spyMedia).toHaveBeenCalledTimes(2);
   });
 
-  it('releases media player on component unmount', () => {
-    const {container, unmount} = render(<MediaPlayer type={'audio'} sources={getAudioSources()} />);
-    const spyMediaRelease = jest.spyOn(media, 'releasePlayer');
-    unmount();
-    expect(spyMediaRelease).toHaveBeenCalled();
-    expect(container.querySelector('audio')).toBeNull();
+  it('calls play on player when shouldPlay changes to true in playerState', () => {
+    let state = getInitialPlayerState();
+    let playerActions = getPlayerActions();
+    const {rerender} = render(<MediaPlayer type={'video'} sources={getVideoSources()} playerState={state} playerActions={playerActions} />);
+    
+    state = getInitialPlayerState();
+    state.shouldPlay = true;
+
+    let spyPlay = jest.spyOn(media.players['0'], 'play');
+    rerender(<MediaPlayer type={'video'} sources={getVideoSources()} playerState={state} playerActions={playerActions} />);
+    expect(spyPlay).toHaveBeenCalledTimes(1);
```

Comment:
It might have been simpler to to stub `media.getPlayer` and return a fake player with mock functions for things like `play`. These tests now depend on the fact that `media` stores its players in a `players` property, which is an implementation detail that might change when we implement the blessed pool.

This is not a blocker, though. We can merge like this and change it later.

Reply:
let's make these changes in blessed player pool task.
So that this gets merged and Daniel can continue.

Reply:
Agreed. I'll merge soon as CI passes.